tests/generated_data/generate_md.py

```diff
@@ -0,0 +1,153 @@
+    "ed25519_2": Key.from_securesystemslib_key(keys_dict["ed25519_2"]),
+    "ed25519_3": Key.from_securesystemslib_key(keys_dict["ed25519_3"]),
+    "ed25519_4": Key.from_securesystemslib_key(keys_dict["ed25519_4"]),
+}
+
+
+class MetadataGenerator:
+    """Responsible for static metadata files generation.
+
+    Args:
+        out_dir: The output dir for the newly generated metadata files.
+    """
+
+    def __init__(self, out_dir: Optional[str] = None) -> None:
+        expires_str = "2050-01-01T00:00:00Z"
+        self.expiry = datetime.strptime(expires_str, "%Y-%m-%dT%H:%M:%SZ")
+        self.dir = out_dir if out_dir else "ed25519_metadata"
+        if not os.path.exists(self.dir):
+            os.mkdir(self.dir)
+
+        self.spec_version = ".".join(SPECIFICATION_VERSION)
+        # Dump metadata files in a human readable format.
+        self.serializer = JSONSerializer()
+
+    def verify_generation(self, md: Metadata, path: str) -> None:
+        """Verify that newly generated file equals the locally stored one.
+
+        Args:
+            md: Newly generated metadata object.
+            path: Path to the locally stored metadata file.
+        """
+        with open(path, "rb") as f:
+            static_md_bytes = f.read()
+            md_bytes = md.to_bytes(self.serializer)
+            if static_md_bytes != md_bytes:
+                raise ValueError(
+                    f"Generated data != local data at {path}. Generate a new "
+                    + "metadata with 'python generated_data/generate_md.py'"
+                )
+
+    def generate_all_files(
+        self, dump: Optional[bool] = False, verify: Optional[bool] = False
+    ) -> None:
+        """Generate a new repository and optionally verify it.
+
+        Args:
+            dump: Wheter to dump the newly generated files inside self.dir.
+            verify: Whether to verify the newly generated files with the
+                local staored.
+        """
+        roles = {role_name: Role([], 1) for role_name in TOP_LEVEL_ROLE_NAMES}
+        root = Root(1, self.spec_version, self.expiry, {}, roles, True)
+        root.add_key("root", keys["ed25519_1"])
+        root.add_key("timestamp", keys["ed25519_2"])
+        root.add_key("snapshot", keys["ed25519_3"])
+        root.add_key("targets", keys["ed25519_4"])
+        md_root: Metadata[Root] = Metadata(root, {})
+
+        timestamp = Timestamp(1, self.spec_version, self.expiry, MetaFile(1))
+        md_timestamp: Metadata[Timestamp] = Metadata(timestamp, {})
+
+        meta: Dict[str, MetaFile] = {"targets.json": MetaFile(1)}
+        snapshot = Snapshot(1, self.spec_version, self.expiry, meta)
+        md_snapshot: Metadata[Snapshot] = Metadata(snapshot, {})
+
+        targets = Targets(1, self.spec_version, self.expiry, {})
+        md_targets: Metadata[Targets] = Metadata(targets, {})
+
+        for md in [md_root, md_timestamp, md_snapshot, md_targets]:
+            assert isinstance(md, Metadata)
```

Comment:
This check is required from mypy otherwise there is a warning:
`
tests/generated_data/generate_md.py:142: error: Argument 1 to "verify_generation" of "MetadataGenerator" has incompatible type "object"; expected "Metadata[Any]"  [arg-type]
`